src/api/entities/SecurityToken/TokenHolders.ts

```diff
@@ -27,16 +42,56 @@ export class TokenHolders extends Namespace<SecurityToken> {
 
     const { entries, lastKey: next } = await requestPaginated(query.asset.balanceOf, {
       arg: rawTicker,
-      paginationOpts,
+      paginationOpts: opts?.pagination || undefined,
     });
 
-    const data: IdentityBalance[] = entries.map(([storageKey, balance]) => ({
-      identity: new Identity(
-        { did: identityIdToString(storageKey.args[1] as IdentityId) },
-        context
-      ),
-      balance: balanceToBigNumber(balance),
-    }));
+    let data: IdentityBalance[] = [];
+
+    const securityToken = new SecurityToken({ ticker }, context);
+
+    const areFrozen = opts?.mintStatus ? await securityToken.transfers.areFrozen() : undefined;
```

Comment:
- You should just default opts to `{ mintStatus: false }` so you can destructure it at the beginning and avoid having to use null conditional operators
- You should use a nested if inside the one below. The case where you requested the mint status and transfers are frozen is different from the case where you didn't request the mint status